packages/sso-menu/src/SSOMenu.tsx

```diff
@@ -0,0 +1,250 @@
+  justify-content: center;
+`;
+
+const menuItemTextStyle = css`
+  font-size: 14px;
+  line-height: 16px;
+  cursor: pointer;
+  margin: 0px;
+`;
+
+const descriptionStyle = css`
+  margin: 0px;
+  display: block;
+  font-weight: normal;
+  font-size: 12px;
+  color: ${uiColors.gray.dark2};
+  text-decoration: none;
+`;
+
+const menuItemPadding = css`
+  padding-left: 20px;
+`;
+
+const logoutContainerHeight = css`
+  height: 46px;
+`;
+
+const menuItems = [
+  {
+    displayName: 'Atlas',
+    description: 'cloud.mongodb.com',
+    href: 'https://cloud.mongodb.com',
+    slug: 'atlas',
+  },
+  {
+    displayName: 'University',
+    description: 'university.mongodb.com',
+    href: 'https://university.mongodb.com',
+    slug: 'university',
+  },
+  {
+    displayName: 'Cloud Support',
+    description: 'support.mongodb.com',
+    href: 'https://support.mongodb.com',
+    slug: 'support',
+  },
+] as const;
+
+export const ActiveProduct = {
+  Atlas: 'atlas',
+  University: 'university',
+  Support: 'support',
+} as const;
+
+type ActiveProduct = typeof ActiveProduct[keyof typeof ActiveProduct];
+
+interface SSOMenuProps {
+  /**
+   * Object that contains information about the active user. {name: 'string', email: 'string'}
+   */
+  user: { name: string; email: string };
+
+  /**
+   * MongoDB product that is currently active: ['atlas', 'university', 'support'].
+   */
+  activeProduct: ActiveProduct;
+
+  /**
+   * Callback invoked when user logs out.
+   */
+  onLogout?: React.MouseEventHandler;
+
+  /**
+   * Callback invoked when user switches products.
+   */
+  onProductChange?: React.MouseEventHandler;
+
+  /**
+   * Callback invoked when user views their MongoDB account.
+   */
+  onAccountClick?: React.MouseEventHandler;
+}
+
+/**
+ * # SSOMenu
+ *
+ * ```
+<SSOMenu
+    user={{ name: 'Alex Smith', email: 'alex.smith@youwork.com' }}
+    activeProduct="atlas"
+    onAccountClick={() => console.log('Viewing account information')}
+    onLogout={() => console.log('On logout')}
+    onProductChange={() => console.log('Switching products')}
+  />
+ * ```
+ * @param props.user Object that contains information about the active user. {name: 'string', email: 'string'}
+ * @param props.activeProduct  MongoDB product that is currently active: ['atlas', 'university', 'support'].
+ * @param props.onLogout Callback invoked when user logs out.
+ * @param props.onAccountClick Callback invoked when user views their MongoDB account.
+ * @param props.onProductChange Callback invoked when user switches products.
+ *
+ */
+function SSOMenu({
+  user: { name, email },
+  activeProduct,
+  onLogout,
+  onProductChange,
+  onAccountClick,
+}: SSOMenuProps) {
+  const triggerRef = useRef<HTMLButtonElement>(null);
+
+  return (
+    <Menu
+      trigger={
+        <button ref={triggerRef} type="button" className={menuButtonStyle}>
+          <span className={menuNameStyle}>{name}</span>
+          <Icon glyph="CaretDown" className={iconStyle} />
```

Comment:
Should we change this to CaretUp when it's opened?

Reply:
I see we're doing that with the `.lg-active`.

How much more work is it to just make this a fully controlled component? Seems like a rule we're breaking that doesn't provide enough value to warrant it.

Reply:
It felt like we were rewriting logic we already wrote about handling backdrop clicks and using escape to close the menu. Can definitely make this a controlled component if need be, but this felt like a good way to make use of the code we've already written. (That's what was driving the decision here)

Reply:
I think I thought Menu behaved like Modal. Where it calls the setClosed function for these events.

Probably worthwhile to port that functionality over to Menu in a separate PR. If we use Menu as a controlled component and then add that functionality to Menu it should 'just work' the way we want.